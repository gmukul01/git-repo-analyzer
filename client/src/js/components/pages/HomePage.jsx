import React, { Component, Fragment } from "react";
import Header from "containers/Header";
import SearchForm from "containers/search/SearchForm";
import SearchResult from "containers/search/SearchResult";

export default class HomePage extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <section className="content">
          <SearchForm />
          <SearchResult />
        </section>
      </Fragment>
    );
  }
}
