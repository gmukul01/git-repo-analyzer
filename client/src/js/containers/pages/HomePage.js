import { connect } from "react-redux";

import HomePage from "components/pages/HomePage";

const mapStateToProps = state => ({
  isUserLoggedIn: state.user ? state.user.name : undefined
});

export default connect(
  mapStateToProps,
  null
)(HomePage);
