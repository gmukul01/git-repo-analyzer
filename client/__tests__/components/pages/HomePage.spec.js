import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import HomePage from "components/pages/HomePage";

describe("HomePage Component", () => {
  const initialProps = {
    location: {
      state: {
        from: "/"
      }
    },
    isUserLoggedIn: true
  };

  it("should match snapshot when user is not logged in", () => {
    const props = {
        ...initialProps,
        isUserLoggedIn: false
      },
      renderer = new ShallowRenderer(),
      result = renderer.render(<HomePage {...props} />);
    expect(result).toMatchSnapshot();
  });

  it("should match snapshot when user is logged in and 'from' is available in the location", () => {
    const renderer = new ShallowRenderer(),
      result = renderer.render(<HomePage {...initialProps} />);
    expect(result).toMatchSnapshot();
  });

  it("should match snapshot when user is logged in and there is no 'from' is available in the location", () => {
    const props = {
        ...initialProps,
        location: {}
      },
      renderer = new ShallowRenderer(),
      result = renderer.render(<HomePage {...props} />);
    expect(result).toMatchSnapshot();
  });
});
