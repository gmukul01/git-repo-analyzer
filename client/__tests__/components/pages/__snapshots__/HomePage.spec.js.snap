// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should match snapshot when user is logged in and 'from' is available in the location 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`HomePage Component should match snapshot when user is logged in and there is no 'from' is available in the location 1`] = `
<React.Fragment>
  <Connect(Header) />
  <section
    className="content home-content"
  >
    <h1>
      This website will give statistics about repos which are hosted on Github
    </h1>
    <Link
      className="btn btn-social btn-github"
      replace={false}
      to="/search"
    >
      <span
        className="fa fa-github"
      />
      Go to Search Page
    </Link>
  </section>
</React.Fragment>
`;

exports[`HomePage Component should match snapshot when user is not logged in 1`] = `
<React.Fragment>
  <Connect(Header) />
  <section
    className="content home-content"
  >
    <h1>
      This website will give statistics about repos which are hosted on Github
    </h1>
    <a
      className="btn btn-social btn-github"
      href="/auth/github"
    >
      <span
        className="fa fa-github"
      />
       Sign in with Github
    </a>
  </section>
</React.Fragment>
`;
